packages/components/src/snackbar/list.js

```diff
@@ -20,20 +27,43 @@ import Snackbar from './';
 * @return {Object}                The rendered notices list.
 */
 function SnackbarList( { notices, className, children, onRemove = noop } ) {
+	const isReducedMotion = useReducedMotion();
+	const [ refMap ] = useState( () => new WeakMap() );
+	const transitions = useTransition(
+		notices,
+		( notice ) => notice.id,
+		{
+			from: { opacity: 0, height: 0 },
+			enter: ( item ) => async ( next ) => await next( { opacity: 1, height: refMap.get( item ).offsetHeight } ),
+			leave: () => async ( next ) => {
+				await next( { opacity: 0 } );
+				await next( { height: 0 } );
+			},
+			immediate: isReducedMotion,
+			config: { tension: 300 },
+		}
+	);
+
 	className = classnames( 'components-snackbar-list', className );
-	const removeNotice = ( id ) => () => onRemove( id );
+	const removeNotice = ( notice ) => () => onRemove( notice.id );
 
 	return (
 		<div className={ className }>
 			{ children }
-			{ notices.map( ( notice ) => (
-				<Snackbar
-					{ ...omit( notice, [ 'content' ] ) }
-					key={ notice.id }
-					onRemove={ removeNotice( notice.id ) }
-				>
-					{ notice.content }
-				</Snackbar>
+			{ transitions.map( ( { item: notice, key, props: style } ) => (
+				<animated.div key={ key } style={ style }>
+					<div
+						className="components-snackbar-list__notice-container"
+						ref={ ( ref ) => ref && refMap.set( notice, ref ) }
+					>
```

Comment:
Why is this an additional nested div?  Could the props and ref just be added to the `<animated.div>` component?

Reply:
No we can't, that was my original ideal but the way this animation work required the animation divs to be adjacent without any margin between them (otherwise the animation is not smooth on removal, there's a jump) and it also requires that the height of the animated div is equal to the snackbar + the margin between two snackbars and without the inner div it's impossible to do because of the "margin-collapsing"